crnk-core/src/main/java/io/crnk/core/boot/CrnkBoot.java

```diff
@@ -417,10 +439,13 @@ public void setServiceDiscovery(ServiceDiscovery serviceDiscovery) {
 	 * NOTE: This using this feature requires a {@link QuerySpecDeserializer} and it does not work with the
 	 * deprecated {@link QueryParamsBuilder}.
 	 */
+	@Deprecated
 	public void setDefaultPageLimit(Long defaultPageLimit) {
 		PreconditionUtil.assertNotNull("Setting the default page limit requires using the QuerySpecDeserializer, but " +
 				"it is null. Are you using QueryParams instead?", this.querySpecDeserializer);
-		((DefaultQuerySpecDeserializer) this.querySpecDeserializer).setDefaultLimit(defaultPageLimit);
+		if (this.querySpecDeserializer instanceof DefaultQuerySpecDeserializer) {
+			((DefaultQuerySpecDeserializer) this.querySpecDeserializer).setDefaultLimit(defaultPageLimit);
```

Comment:
potentially it should rather throw an exception if it is not possible to set the page limit